go-selinux/selinux_linux.go

```diff
@@ -1095,3 +1104,115 @@ func dupSecOpt(src string) ([]string, error) {
 func disableSecOpt() []string {
 	return []string{"disable"}
 }
+
+func getDefaultContextFromReaders(c defaultSECtx) (string, error) {
+	if c.verifier == nil {
+		return "", fmt.Errorf("verifier function must not be nil")
+	}
+
+	context, err := newContext(c.scon)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create label for %s", c.scon)
+	}
+
+	findUserInContext := func(r io.Reader) string {
+		fromRole := context["role"]
+		fromType := context["type"]
+		bufin := bufio.NewReader(r)
+
+		for done := false; !done; {
+			var (
+				line string
+				err  error
+			)
+
+			if line, err = bufin.ReadString('\n'); err != nil {
+				if err == io.EOF {
+					done = true
+				} else {
+					break
+				}
+			}
+			line = strings.TrimSpace(line)
+			if len(line) == 0 {
+				// Skip blank lines
+				continue
+			}
+			if line[0] == ';' || line[0] == '#' {
+				// Skip comments
+				continue
+			}
+
+			conns := strings.Fields(line)
```

Comment:
Since you have used TrimSpace already, it might make sense to use strings.Split here (it's 2x faster than strings.Fields).

If your input is more complicated (various delimiters, may have 2 spaces instead of one etc.), use `strings.Fields` but in this case you don't need `strings.TrimSpace` beforehand.

Reply:
>  it might make sense to use strings.Split here (it's 2x faster than strings.Fields).
Is there recommended reading you can point me to about performant Go? Or is this something you just know from experience? Thanks